commit 11f06c02e36af4dc330b9d8c651ec50c7b131984
Author: Marek Dajnowski <fornve@yahoo.co.uk>
Date:   Thu Dec 31 12:35:59 2009 +0000

    Joomla! 1.6 DAO patch

diff --git a/libraries/joomla/database/database.php b/libraries/joomla/database/database.php
index 160e5b3..43bac44 100644
--- a/libraries/joomla/database/database.php
+++ b/libraries/joomla/database/database.php
@@ -671,19 +671,21 @@ abstract class JDatabase extends JObject
 	/**
 	 * This global function loads the first row of a query into an object
 	 *
+	 * @param	string	Class name of object returned
 	 * @return	object
 	 */
-	abstract public function loadObject();
+	abstract public function loadObject( $class_name = 'stdClass' );
 
 	/**
 	 * Load a list of database objects
 	 *
 	 * @param	string	The field name of a primary key
+	 * @param	string	Class name of object returned
 	 * @return	array	If <var>key</var> is empty as sequential list of returned records.
 	 * If <var>key</var> is not empty then the returned array is indexed by the value
 	 * the database key.  Returns <var>null</var> if the query fails.
 	 */
-	abstract public function loadObjectList($key='');
+	abstract public function loadObjectList($key='', $class_name = 'stdClass' );
 
 	/**
 	 * Load the first row returned by the query
diff --git a/libraries/joomla/database/database/mysql.php b/libraries/joomla/database/database/mysql.php
index afec27b..7627823 100644
--- a/libraries/joomla/database/database/mysql.php
+++ b/libraries/joomla/database/database/mysql.php
@@ -407,15 +407,16 @@ class JDatabaseMySQL extends JDatabase
 	/**
 	* This global function loads the first row of a query into an object
 	*
+	* @param	string	Class name of object returned
 	* @return 	object
 	*/
-	public function loadObject()
+	public function loadObject( $class_name = 'stdClass' )
 	{
 		if (!($cur = $this->query())) {
 			return null;
 		}
 		$ret = null;
-		if ($object = mysql_fetch_object($cur)) {
+		if ($object = mysql_fetch_object($cur, $class_name )) {
 			$ret = $object;
 		}
 		mysql_free_result($cur);
@@ -428,16 +429,17 @@ class JDatabaseMySQL extends JDatabase
 	* If <var>key</var> is not empty then the returned array is indexed by the value
 	* the database key.  Returns <var>null</var> if the query fails.
 	*
-	* @param	string	The field name of a primary key
+	* @param	string	The field name of a primary key  
+	* @param	string	Class name of objects returned
 	* @return	array	If <var>key</var> is empty as sequential list of returned records.
 	*/
-	public function loadObjectList($key='')
+	public function loadObjectList($key='', $class_name = 'stdClass' )
 	{
 		if (!($cur = $this->query())) {
 			return null;
 		}
 		$array = array();
-		while ($row = mysql_fetch_object($cur))
+		while ($row = mysql_fetch_object($cur, $class_name ))
 		{
 			if ($key) {
 				$array[$row->$key] = $row;
diff --git a/libraries/joomla/database/database/mysqli.php b/libraries/joomla/database/database/mysqli.php
index 65ca7d1..8adc266 100644
--- a/libraries/joomla/database/database/mysqli.php
+++ b/libraries/joomla/database/database/mysqli.php
@@ -429,15 +429,16 @@ class JDatabaseMySQLi extends JDatabase
 	/**
 	 * This global function loads the first row of a query into an object
 	 *
+	 * @param	string	Class name of object returned
 	 * @return	object
 	 */
-	public function loadObject()
+	public function loadObject( $class_name = 'stdClass' )
 	{
 		if (!($cur = $this->query())) {
 			return null;
 		}
 		$ret = null;
-		if ($object = mysqli_fetch_object($cur)) {
+		if ($object = mysqli_fetch_object($cur, $class_name = 'stdClass' )) {
 			$ret = $object;
 		}
 		mysqli_free_result($cur);
@@ -451,15 +452,16 @@ class JDatabaseMySQLi extends JDatabase
 	 * the database key.  Returns <var>null</var> if the query fails.
 	 *
 	 * @param	string	The field name of a primary key
+	 * @param	string	Class name of objects returned
 	 * @return	array	If <var>key</var> is empty as sequential list of returned records.
 	 */
-	public function loadObjectList($key='')
+	public function loadObjectList($key='', $class_name = 'stdClass' )
 	{
 		if (!($cur = $this->query())) {
 			return null;
 		}
 		$array = array();
-		while ($row = mysqli_fetch_object($cur))
+		while ($row = mysqli_fetch_object($cur, $class_name ))
 		{
 			if ($key) {
 				$array[$row->$key] = $row;
